client/src/components/project-selection/WorkspaceSelector.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState } from "react";
+import { FormControl, InputLabel, MenuItem, Select } from "@mui/material";
+import WorkspaceItem from "./WorkspaceItem";
+import NewWorkspaceModal from "./NewWorkspaceModal";
+import EditWorkspaceModal from "./EditWorkspaceModal";
+import NewWorkspaceItem from "./NewWorkspaceItem";
+import { DeleteWorkspaceModal } from "./DeleteWorkspaceModal";
+import { useWorkspaceStore, workSpaceType } from "../../stores/workspaceStore";
+
+export type openModal = "create" | "edit" | "delete" | "none";
+const emptyWorkSpace: workSpaceType = {
+  id: "",
+  name: "",
+  isOpen: false,
+};
+const ISHIDDEN: boolean = true;
+const WorkspaceSelector = () => {
+  const [modalIsOpen, setModalIsOpen] = useState<openModal>("none");
```

Comment:
use booleans and not strings.

Reply:
But i have 4 options:
Create, delete, edit, none.
This is instead of making for every one of them isOpen boolean state

Reply:
Ok got your, in this case, you should pass booleans to the modal based on the string.
```
<EditWorkspaceModal
            modalIsOpen={modalIsOpen === "edit"}
            onCancel={handleModalIsClosed}
            workSpace={workSpaceToEdit}
          />
```